tests/contracts/blsContracts/BLSLibrary.sol

```diff
@@ -0,0 +1,235 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.26;
+
+import "./BigNumber.sol";
+import "./Elements.sol";
+
+library BLSLibrary {
+    using BigNumbers for *;
+
+    bytes constant moduloBytes =
+        hex"1a0111ea397fe69a4b1ba7b6434bacd764774b84f38512bf6730d2a0f6b0f6241eabfffeb153ffffb9feffffffffaaab";
+    
+    // Negated generator of G1 to be used for pairing function
+    bytes constant negG1 =
+        hex"0000000000000000000000000000000017f1d3a73197d7942695638c4fa9ac0fc3688c4f9774b905a14e3a3f171bac586c55e83ff97a1aeffb3af00adb22c6bb00000000000000000000000000000000114d1d6855d545a8aa7d76c8cf2e21f267816aef1db507c96655b9d5caac42364e6f38ba0ecb751bad54dcd6b939c2ca";
+
+    // This defines how many fp elements have to be created when hashing message
+    // For EncodeToG2 this is 2
+    // For HashToG2 this is 2*2
+    uint16 constant elCount = 2;
+
+    uint16 private constant L = 64;
+
+    // Domain separation tag 
+    // TODO: need to be specified according to spec recommendation
+    bytes constant dst = "";
+
+    // Main function to hash message into G2 point
+    function EncodeToG2(bytes memory message)
+        public
+        view
+        returns (bytes memory)
+    {
+        Elements.Element[] memory elArray = hashToFieldPoints(message);
+        bytes memory fpBytes;
+        uint64 zero;
+        for (uint8 i; i < elArray.length; i++) {
+            fpBytes = abi.encodePacked(
+                fpBytes,
+                zero,
+                zero,
+                elArray[i].val[5],
+                elArray[i].val[4],
+                elArray[i].val[3],
+                elArray[i].val[2],
+                elArray[i].val[1],
+                elArray[i].val[0]
+            );
+        }
+        return precompileMaptoG2(fpBytes);
+    }
+
+    // All input parameters are points on G1 or G2 as bytes
+    function Pair(
```

Comment:
How about renaming this to "CheckSignature" or "IsValidSignature" if you want to stress that the function returns a boolean and does not revert the contract in case of a failure?

The function is internally using the `pair` function to check the the signature, but it is not actually returning the result of the pairing.